Engine/AlgorithmManager.cs

```diff
@@ -997,5 +989,19 @@ private void SampleBenchmark(IAlgorithm algorithm, IResultHandler results, DateT
                 Log.Error(err);
             }
         }
+
+        /// <summary>
+        /// Determines if a data point is in it's native, configured resolution
+        /// </summary>
+        private static bool EndTimeIsInNativeResolution(SubscriptionDataConfig config, DateTime dataPointEndTime)
+        {
+            if (config.Increment == TimeSpan.Zero)
+            {
+                return true;
+            }
+
+            var roundedDataPointEndTime = dataPointEndTime.RoundDownInTimeZone(config.Increment, config.ExchangeTimeZone, config.DataTimeZone);
```

Comment:
I'm not 100% sure how this works but: During an execution, if this is true once, wouldn't it be true always for that data source? or can it change?
If you can trust it will return always the same, maybe there could be some type of cache/memory of previous checks, to avoid calling RoundDownInTimeZone for each data.

Reply:
the date time value here will be different for every data point so we can't cache the value